### ISSUE TITLE
Simple console sender sends thousands of messages if terminated 

### ISSUE THREAD

**Issue by Loreno10** [Azure/azure-iot-sdk-csharp#542]
- **OS, version, SKU and CPU architecture used:** Windows 7 Pro x64
- **Application's .NET Target Framework :** netcoreapp2.0
- **Device:** Laptop
- **SDK version used:** Microsoft.Azure.Devices.Client 1.7.1

## Description of the issue:
I have a simple sender application, which I did following the official tutorial on MSDN. I observed that after closing the console application, my IoTHub gets a really big amount of D2C messages in a short time. After a few minutes I see in portal that I have more than 9000 messages (where my free IoTHub instance handles only 8000). What's the cause?
I'm thinking that maybe I should call DeviceClient.CloseAsync() before closing the application. Would it help? I wish to find out myself, but as I said, my messages limit run out for today. Before I experiment next time, better to ask here.

I also observed that even after my app is closed, in Windows processes dotnet.exe is still running - why is that?

## Code sample exhibiting the issue:
Here's how I run my app:
I created a bash script, which I run with Git Bash:

> #!/bin/bash
> cd /c/apps/app1/app1/bin/Debug/netcoreapp2.0
> dotnet app1.dll

Main loop of my app:

```
            while (true)
            {
                Console.Write("New message: ");
                var choice = Console.ReadLine();

                if (choice == " ")
                    choice = GetRandomValue(format, choice);

                Console.WriteLine($"Sending: {choice}");

                var message = GetMessage(_device.DeviceId, propertyName, choice);
                await sender.SendEventAsync(message);
            }
```

**Comment by CIPop** [comment]
> I observed that after closing the console application, my IoTHub gets a really big amount of C2D messages in a short time.

@Loreno10 I'm not sure I understand the question: it appears that you're sending Device To Cloud messages, not Cloud to Device messages in a tight loop, as long as you press the Enter key.
Looking only at the above code, this should create a number of messages equal to the number of times you press Enter, regardless if you call CloseAsync or not. 

While we do recommend CloseAsync to terminate connections gracefully, once the process is terminated, there isn't anything that could continue sending.

If you consistently see this happening with your free hub and you can verify there is absolutely no other way you are sending messages (e.g. not awaiting async operations, etc) then please open a support ticket from within your subscription so we can check the logs from the service side. (See [this ](https://github.com/azure/azure-iot-sdk-csharp#need-support) for how to open an SR.)

**Comment by Loreno10** [comment]
@CIPop Sorry, I meant D2C of course.

You are right, message should be sent only when ENTER is pressed, but for some reason that I don't understand, after terminating the app, it sends thousands of messages.

I think it has something to do with the fact that dotnet.exe process is staying in the background after the app is closed.

I setup on my sender a method invocation routine, so I can invoke a sender's method from another application. I noticed that even after closing the application I am still able to fire this method - which means the device is online.

So, basically two things are problematic: dotnet.exe not terminating and my app magically sending thousands of messages without pressing ENTER.

**Comment by CIPop** [comment]
Closing for now - this appears to be an application issue.

**Comment by Loreno10** [comment]
I have no idea how to attach a debugger to application that I do not run from Visual Studio, but thanks anyway. I'll be just closing the application from Processes window - when I do that, there is no unexpected behaviour.

**Comment by CIPop** [comment]
@Loreno10 : this might help: https://msdn.microsoft.com/en-us/library/3s68z0b3.aspx 
If you built the app as "Debug" you should be able to simply load the sln in Visual Studio then attach.
